Add CLI command reference docs page; update API Services docs page; add create/fetch docs page with more sample files and examples (#1546)
Axway/axway-open-docs: content/en/docs/central/cli_central/cli_command_reference.md

---
title: AMPLIFY Central CLI Command Index
linkTitle: AMPLIFY Central CLI Command Index
weight: 130
date: 2021-01-13T00:00:00.000Z
description: Learn about the different AMPLIFY Central CLI commands you can use
---

## **get**

List one or more resources.

Prints a table of the most important information about the specified resources.

If the desired resource type is scoped you will need to specify the scope name by providing `-s, --scope <name>` flag.

You can query for more than one resource if using comma-separated resources in the command (see the examples).

Querying for multiple resources will display multiple result tables, one result table for each resource you fetch.

> To see the list of all available resources on the server run `amplify central get`

Reviewer: Prefer not to mention "server".
```suggestion
To see the list of all available resources from AMPLIFY Central, run `amplify central get`
```

---
title: AMPLIFY Central CLI Command Index
linkTitle: CLI commands Reference
weight: 130
date: 2021-01-13T00:00:00.000Z
description: Learn about the different AMPLIFY Central CLI commands you can use.
---

This section contains the basic commands for creating, fetching, updating, and deleting various Axway API Server assets using the Amplify Central CLI. Each command is followed by a brief description of what the command does, an explanation of the proper command syntax including command arguments and options, along with example syntax for various use-cases. Use this page to quickly lookup more information about the Amplify Central CLI command you're utilizing.

## get

> This command lists one or more resources. It also prints a table of the most important information about an specified resources. If the desired resource type is scoped, you will need to specify the scope name by providing the `-s, --scope <name>` flag.

Reviewer: Should we explain the meaning of a scoped resource versus resources that are not scoped with examples?

|`amplify central get <Resource>`                               |# Get a list of the resources          |
|`amplify central get <Resource1>,<Resource2>,...,<ResourceN>`  |# Get a list of multiple resources  |
|`amplify central get <Resource> <Name> -s/--scope <Scope Name>`|# Get a specific resource by name |
|**Options**                                                    |                   |
|`--client-id=<value>`                                          |# Override your DevOps account's client ID |
|`-o,--output=<value>`                                          |# Additional output formats, yaml or json  |
|`-s,--scope=<name>`                                            |# Scope name for scoped resources          |
|**Arguments**                                                    |                   |
|args...                                                        |# Command arguments, run `amplify central get` to see the examples |

The following are examples of how to use the `get` command:

To get all environments:

```
amplify central get envs
```

To get all environments in yaml format

```
amplify central get environments -o yaml
```

> To get environment by Resource/Common name in json format

Reviewer: To make it relevant to the example provided.    Suggest saying edit to say  "Resource/Common name (i.e. myenv)"

```
amplify central get env myenv -o json
```

To get all webhooks

```
amplify central get webhooks
```

To get all webhooks by Short Name

```
amplify central get webh
```

To get all webhooks and apiservices by Short Name

```
amplify central get webh,apis
```

To get all environments and API services

```
amplify central get envs,apisvc
```

> To get an environment and an apiservice which matches a Resource/Common Name in a specified scope (scope is required after -s flag) in json format

Reviewer: Suggest edit to say "(scope is env1 in the example and required after -s flag)"

### Edit Usage

`amplify central edit <command> [options]`

### Edit Commands

  `environment`   Edit an environment with the specified name.

### Edit Options

  `--client-id=<value>`   Override your DevOps account's client ID

  `-o,--output=<value>`   Additional output formats. One of: yaml|json

#### **edit: resource by name**

Edit an environment with the specified name.

_Using "environment" as an example._

Usage:

`amplify central edit environment [options] <name>`

Environment Arguments:

  `name`          Name of the environment

### Edit Examples

```bash
# delete environment by name
amplify central edit environment newenv
```

- - -

## **install**

Install additional platform resources

### Install Usage

`amplify central install <command> [options] [<args...>]`

### Install Commands

  `agents`                Install v7 / AWS / kubernetes agent.

### Install Arguments

  `args...`               Command arguments, run `amplify central install` to see the examples

### Install Options

`--client-id=<value>`   Override your DevOps account's client ID

`--region=<value>`      Override your region config

#### **install: agents configuration**

> Install API Gateway v7 / Amazon API Gateway / Kubernetes agents

Reviewer: I asked Cyrille if this needs to include new parameter for 'azure' agents in the CLI.  Meaning has it been done yet.

Reviewer: No need to change to mention Azure yet.   Its not a support optional parameter yet.

Reviewer: Correction, the install agents will support Azure this week.
![image](https://user-images.githubusercontent.com/51344789/105877128-de1f0f80-5fbc-11eb-9c24-2bb907f55954.png)

---
title: AMPLIFY Central CLI Command Index
linkTitle: CLI commands Reference
weight: 130
date: 2021-01-13T00:00:00.000Z
description: Learn about the different AMPLIFY Central CLI commands you can use.
---

This section contains the basic commands for creating, fetching, updating, and deleting various Axway API Server assets using the Amplify Central CLI. Each command is followed by a brief description of what the command does, an explanation of the proper command syntax including command arguments and options, along with example syntax for various use-cases. Use this page to quickly lookup more information about the Amplify Central CLI command you're utilizing.

## get

This command lists one or more resources. It also prints a table of the most important information about an specified resources. 

> Resources can be *scoped* or *unscoped*. A resource's scope refers to the accessibility of the resource. A scoped resource type will be deleteed when its scoped/related resource is deleted. Access to that resource type is controlled by authorization checks on the scope. An *unscoped* resource's availability/accessibility is *independent* of the accessibility of other resources.

Author: @robertengaxway I added this explanation of scoped vs unscoped. Let me know if it's ok or if I should revise it.

## get

This command lists one or more resources. It also prints a table of the most important information about an specified resources. 

> Resources can be *scoped* or *unscoped*. A resource's scope refers to the accessibility of the resource. A scoped resource type will be deleteed when its scoped/related resource is deleted. Access to that resource type is controlled by authorization checks on the scope. An *unscoped* resource's availability/accessibility is *independent* of the accessibility of other resources.

Reviewer: typo on 'deleteed', should be 'deleted'

> To see the list of all available resources on the server run `amplify central delete`.

Reviewer: replace "server" with "API Server" as used in https://docs.axway.com/bundle/axway-open-docs/page/docs/central/integrate_with_webhooks/index.html 
We can use the formal "Axway API Server" in the intro section of the Command Summary or the Central CLI intro.

Author: @robertengaxway I made the suggested change on that line(server --> API Server), thanks
